sping.go

```diff
@@ -14,12 +16,18 @@ func init() {
 }
 
 // NewClient returns a ping client with the specified options.
-func NewClient(name string, delay int64, limit uint) *PingClient {
+func NewClient(dailer Dailer, address, name string, delay int64, limit uint) *PingClient {
```

Comment:
Ok, so your proposed interface is that you pass a dialer to the `NewClient` function and the dialers are defined in the code. Any reason this is preferred over making the dialers methods of the client?